### ISSUE TITLE
Remove jwt-decode as a peerDependency

### ISSUE THREAD

**Issue by pirelenito** [klarna/higher-order-components#45]
We should extract the component that uses it on its own package. 

Wdyt @deepsweet @xaviervia ?

**Comment by pirelenito** [comment]
Another option wolud be to make it as a dependency (I looked and it is not that big), ship the package as **ES2015 modules** and let clients deal with not being bundle by simple tree-shaking.

**Comment by deepsweet** [comment]
monorepo

![](http://funkyimg.com/i/2vYE9.jpg)

**Comment by pirelenito** [comment]
Yep! That is even better... mono repo the shit out of these hocs so that they are published individually.

**Comment by deepsweet** [comment]
as an example https://github.com/deepsweet/hocs

**Comment by pirelenito** [comment]
Fantastic! Should we make this happen?

**Comment by deepsweet** [comment]
Does Sagui support it as well as multiple builds?

```
node_modules/@hocs/with-lifecycle
├── dist
│   ├── with-lifecycle.js
│   └── with-lifecycle.min.js
├── es
│   └── index.js
├── lib
│   └── index.js
└── package.json
```

**Comment by pirelenito** [comment]
But we lerna, it would be basically multiple packages, each with their own `package.json` \ `sagui`.

But given these are just libraries, we can also just use Babel and Jest directly. We don't need the webpack for anything anyway.

**Comment by deepsweet** [comment]
Well, React itself and many ecosystem packages transpiles to `dist/` folder with Webpack as well to be able to use it on JS Bin etc.

```
node_modules/recompose/build/
├── Recompose.js
└── Recompose.min.js
```

**Comment by deepsweet** [comment]
Actually the right name for this folder is `umd`.

**Comment by pirelenito** [comment]
Gotcha, we can do that as well.

**Comment by pirelenito** [comment]
Either way, this sounds like a "major" endeavor, and I need the `jwt-decode` fixed ASAP.

I'll prepare a first PR that simply adds it as an official dependency, and we can do the monorepo as a propper long term fix.

Sound good?

**Comment by deepsweet** [comment]
Good.

**Comment by xaviervia** [comment]
Let’s do it!

🐒repo!

(🐒 is "mono" in Spanish)

**Comment by pirelenito** [comment]
btw, github.com/hocs is not taken... kinda like it is wanting for a org to rise on it!

**Comment by xaviervia** [comment]
But it can be a the mono repo of the org!

**Comment by xaviervia** [comment]
Imagine that, we can have an org with a mono repo that repos all the monos!

**Comment by deepsweet** [comment]
and there will be only one repo called `hocs`, so `github.com/hocs/hocs`. yo dawg.

**Comment by xaviervia** [comment]
_exactly_
